syft/execution/role.py

```diff
@@ -295,15 +293,24 @@ def detail(worker: AbstractWorker, role_tuple: "Role") -> tuple:
         for ph in state.state_placeholders:
             placeholders[ph.id.value] = ph
 
-        return Role(
+        role = Role(
             id=id_,
             owner=worker,
             actions=actions,
             state=state,
-            placeholders=placeholders,
+            # placeholders=placeholders,
             input_placeholder_ids=input_placeholder_ids,
             output_placeholder_ids=output_placeholder_ids,
         )
+        for ph in placeholders.values():
+            ph.role = role
+        for ph in state.state_placeholders:
+            ph.role = role
+
+        role.placeholders = placeholders
+        role.state = state
```

Comment:
```suggestion
```
If they're passed in to the `__init__`, seems like these are redundant. 🤔

Reply:
My problem here is that placeholders have a role and roles have placeholders so on detail, I could have infinite recursion if I don't take care.
My solution (until we find a better one) was to not detail the role of a placeholder on `Placeholder.detail()` but detail the placeholders of a Role and set their role to the current role being detailed.